### ISSUE TITLE
bad argument #1 to 'receive'(bad pattern argument)

### ISSUE THREAD

**Issue by linlihuiyang** [pantsel/kong-middleman-plugin#24]
access.lua:94  bad argument #1 to 'receive'(bad pattern argument)

**Comment by tushard-iprogrammercom** [comment]
@linlihuiyang  have u been able to resolve it? We are facing the same issue.

**Comment by tushard-iprogrammercom** [comment]
@linlihuiyang Through further debugging, this line tries to read the 'Content-Length' header from the response. And due to some reason, this header was removed. For us, the reason was a server-side monitoring tool 'Dynatrace'. It is adding cookie header to the original response and thus content-length is removed probably.

**Comment by tushard-iprogrammercom** [comment]
@pantsel are there any other ways to read the response body without using Content-Length header in TCP socket connection?
For now, as we can not remove the Dynatrace from client's setup, we have modified the access.lua to make an HTTP connection with the service. Please, if possible, guide us on if there are any risks in using HTTP instead of TCP.
FYI - we have tried `receive("*a")` - it keeps the connection alive till the timeout and fails at browser due to timeout itself.
we have tried `receive("*l")` - with a different regex as well, but it breaks after it has done reading the headers, not sure why.

**Comment by abdulrazakbg** [comment]
@tushard-iprogrammer @linlihuiyang . Its very old thread. have you got any solution for this ?

**Comment by tushard-iprogrammercom** [comment]
Yeah. I had to rewrite some of the files. Basically, I rewrote them to use
HTTP connection instead of TCP one.

And the main reason was:
- LUA was not having any direct method to read response body and headers
separately.
- So, I had to rely on regex based logic to identify if the header reading
is completed from the buffered response, and after headers are completed,
it reads body by using content-length parameter
- And that logic was breaking due to - some N/w monitoring tool named
"dynatrace" was adding headers with special characters
- With HTTP connection I could read the body directly which was all I
needed.

Regards,
------------------------------------------------

Tushar Dahiwale
Senior Software Engineer
***@***.***

iProgrammer Solutions Pvt. Ltd.
Office 103, 104, 1st Floor Pride Portal,
Shivaji Housing Society, Bahiratwadi,
Near Hotel JW Marriott, Off Senapati Bapat Road,
Pune - 411016, MH, INDIA.
Phone: +91 9595456505 | Skype: tushard_iprogrammer
www.iprogrammer.com  <http://www.iprogrammer.com/>
------------------------------------------------
